packages/sanity/src/core/components/collapseMenu/CollapseMenu.tsx

```diff
@@ -209,45 +297,74 @@ export const CollapseMenu = forwardRef(function CollapseMenu(
                     {...tooltipProps}
                   >
                     <Flex>
-                      {cloneElement(child, {
-                        disabled: child.props.disabled || hidden,
+                      {cloneElement(optionElement, {
+                        disabled: optionElement.props.disabled || hidden,
                         'aria-hidden': hidden,
                       })}
                     </Flex>
                   </Tooltip>
+                </React.Fragment>
+              )
+            })}
+        </RowFlex>
+        {/* Rendered hidden in order to calculate intersections for expanded menu options */}
+        <RowFlex data-hidden aria-hidden="true" gap={gap} overflow="hidden">
```

Comment:
I suggest unifying the rendering of the hidden rows in a function called e.g `renderHiddenRow` that receives an array of items (`React.ReactElement[]`) and an `onIntersectionChange` callback. 

```ts
  const renderHiddenRow = useCallback(
    (
      items: ReactElement[],
      onIntersectionChange: (e: IntersectionObserverEntry, child: React.ReactElement) => void,
    ) => {
      return (
        <RowFlex data-hidden aria-hidden="true" gap={gap} overflow="hidden">
          {items.map((child, index) => {
            const {dividerBefore} = child.props
            return (
              <React.Fragment key={child.key}>
                {dividerBefore && index !== 0 && <CollapseMenuDivider hidden />}

                <OptionObserveElement
                  options={intersectionOptions}
                  // eslint-disable-next-line react/jsx-no-bind
                  onIntersectionChange={(e) => onIntersectionChange(e[0], child)}
                >
                  <Flex>
                    {cloneElement(child, {
                      disabled: true,
                      'aria-hidden': true,
                    })}
                  </Flex>
                </OptionObserveElement>
              </React.Fragment>
            )
          })}
        </RowFlex>
      )
    },
    [gap, intersectionOptions],
  )

```

And then do this in the JSX:

```ts
function AutoCollapseMenu() {
  // ... rest of code

  const renderHiddenRow = useCallback(() => {
    // Return the hidden row
  }, [])

  return (
    <OuterFlex>
      <RootFlex>
        <RowFlex>
          {/* The visible row here */}

          {/* Rendered hidden in order to calculate intersections for expanded menu options */}
          {renderHiddenRow(menuOptions, handleExpandedIntersection)}
          
          {/* Rendered hidden in order to calculate intersections for collapsed menu options */}
          {renderHiddenRow(collapsedElements, handleCollapsedIntersection)}
        </RowFlex>
      </RootFlex>
    </OuterFlex>
  )
}
```

Reply:
Thanks @hermanwikner. That's is a great suggestion! I implemented a slight modification of it (using a component instead of inline function). See ca6f97f67d21ea756b5f426b3e1de4dfaab387ac